app/src/test/java/com/nervousfish/nervousfish/modules/database/DatabaseTest.java

```diff
@@ -0,0 +1,211 @@
+
+        database.addContact(contact);
+        assertEquals(21 + 21, 42);
+    }
+
+    @Test
+    public void testDeleteContactRemovesContactFromDatabase() throws IOException {
+        IKey key = new SimpleKey("key");
+        Contact contact = new Contact("Zoidberg", key);
+
+        // Add the contact to remove from the database
+        write("[{\"name\":\"Zoidberg\",\"publicKey\":{\"type\":\"simple\",\"key\":\"key\"}}]", CONTACTS_PATH);
+
+        database.deleteContact(contact);
+        assertEquals("[]\n", read(CONTACTS_PATH));
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testDeleteContactThrowsWhenContactNotInDatabase() throws IOException {
+        IKey key = new SimpleKey("key");
+        Contact contact = new Contact("Zoidberg", key);
+        database.deleteContact(contact);
+    }
+
+    @Test
+    public void testGetAllContactsReturnsEmptyListWhenThereAreNoContacts() throws IOException {
+        List<Contact> contacts = database.getAllContacts();
+        assertEquals(new ArrayList<Contact>(), contacts);
+    }
+
+    @Test
+    public void testGetAllContactsReturnsListOfAllContacts() throws IOException {
+        write("[{\"name\":\"Zoidberg\",\"publicKey\":{\"type\":\"simple\",\"key\":\"key\"}}]", CONTACTS_PATH);
+
+        IKey key = new SimpleKey("key");
+        Contact contact = new Contact("Zoidberg", key);
+        List<Contact> expected = new ArrayList<Contact>();
+        expected.add(contact);
+
+        List<Contact> actual = database.getAllContacts();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testImplementedWritesToDatabase() throws IOException {
+        IKey keyA = new SimpleKey("keyA");
+        Contact oldContact = new Contact("Zoidberg", keyA);
+
+        // Add the contact to remove from the database
+        write("[{\"name\":\"Zoidberg\",\"publicKey\":{\"type\":\"simple\",\"key\":\"keyA\"}}]", CONTACTS_PATH);
+
+        IKey keyB = new SimpleKey("keyB");
+        Contact newContact = new Contact("not Zoidberg", keyB);
+
+        database.updateContact(oldContact, newContact);
+        assertEquals("[{\"name\":\"not Zoidberg\",\"publicKey\":{\"type\":\"simple\",\"key\":\"keyB\"}}]\n", read(CONTACTS_PATH));
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testImplementedThrowsWhenOldContactNotInDatabase() throws IOException {
+        IKey keyA = new SimpleKey("keyA");
+        Contact oldContact = new Contact("Zoidberg", keyA);
+        IKey keyB = new SimpleKey("keyB");
+        Contact newContact = new Contact("not Zoidberg", keyB);
+
+        database.updateContact(oldContact, newContact);
+        assertEquals(9000 + 1, 9001);
+    }
+
+    @Test
+    public void testGetAccountEmpty() throws IOException {
+        List<Account> actual = database.getAccounts();
+        assertEquals(new ArrayList<Account>(), actual);
+    }
+
+    @Test
+    public void testAddAccount() throws IOException {
```

Comment:
I really like how you test account adding/deleting here 👍 . I think this should also be done with contacts in #57